dataprofiler/tests/profilers/test_unstructured_text_profile.py

```diff
@@ -0,0 +1,141 @@
+import unittest
+from unittest import mock
+from collections import defaultdict
+
+import pandas as pd
+
+from dataprofiler.profilers.unstructured_text_profile import TextProfiler
+
+class TestUnstructuredTextProfile(unittest.TestCase):
+    
+    
+    def test_text_profile_update_and_name(self):
+        text_profile = TextProfiler("Name")
+        sample = pd.Series(["Hello my name is Grant",
+                            "Bob and Grant are friends"])
+        text_profile.update(sample)
+        self.assertEqual("Name", text_profile.name)
+
+        
+    def test_vocab(self):
+        text_profile = TextProfiler("Name")
+        sample = pd.Series(["Hello my name is Grant",
+                            "Bob and Grant are friends"])
+        text_profile.update(sample)
+        profile = text_profile.profile
+
+        # Assert vocab is correct
+        expected_vocab = ['y', 'e', 'G', 'H', 'r', 'b', 'l', 's', 'm',
+                          't', 'i', 'B', 'd', 'f', 'o', ' ', 'n', 'a']
+        self.assertListEqual(sorted(expected_vocab), sorted(profile['vocab']))
+
+        # Update the data again
+        sample = pd.Series(["Grant knows how to code",
+                            "Grant will code with Bob"])
+        text_profile.update(sample)
+        profile = text_profile.profile
+
+        # Assert vocab is correct
+        expected_vocab = [' ', 'B', 'G', 'H', 'a', 'b', 'c', 'd', 'e',
+                          'f', 'h', 'i', 'k', 'l', 'm', 'n', 'o', 'r',
+                          's', 't', 'w', 'y']
+        self.assertListEqual(sorted(expected_vocab), sorted(profile['vocab']))
+
+ 
+    def test_words_and_word_count(self):
+        text_profile = TextProfiler("Name")
+        sample = pd.Series(["Hello my name is Grant",
+                            "Bob and Grant are friends"])
```

Comment:
can we add some punctuations to these tests?

Reply:
Good idea